tb_plugin/torch_tb_profiler/io/file.py

```diff
@@ -0,0 +1,851 @@
+import glob as py_glob
+import os
+import tempfile
+from abc import ABC, abstractmethod
+from collections import namedtuple
+
+from .. import utils
+from .utils import as_bytes, as_text
+
+logger = utils.get_logger()
+
+try:
+    import boto3
+    import botocore.exceptions
+
+    S3_ENABLED = True
+except ImportError:
+    S3_ENABLED = False
+
+try:
+    from azure.storage.blob import ContainerClient
+    BLOB_ENABLED = True
+except ImportError:
+    BLOB_ENABLED = False
+
+_DEFAULT_BLOCK_SIZE = 16 * 1024 * 1024
+
+# Registry of filesystems by prefix.
+#
+# Currently supports "s3://" URLs for S3 based on boto3 and falls
+# back to local filesystem.
```

Comment:
It supports Azure as well. You should mention that in your comment (is this just copied from somewhere else?)